File: src/main/java/com/softwareplace/security/authorization/ResponseRegister.java
package com.softwareplace.security.authorization;

import java.io.IOException;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

import javax.servlet.http.HttpServletResponse;

import com.fasterxml.jackson.databind.ObjectMapper;

public class ResponseRegister {

	private ResponseRegister() {
	}

	public static void register(HttpServletResponse response) throws IOException {
		if (response.getStatus() == HttpServletResponse.SC_UNAUTHORIZED) {
			register(response, "Access denied!", response.getStatus(), new HashMap<>());
		}
	}

	public static void register(HttpServletResponse response, String message, int status, Map<String, Object> params) throws IOException {
		message = message == null ? "Unexpected Error" : message;
		HashMap<String, Object> responseParams = new HashMap<>();
		responseParams.put("message", message);
		responseParams.put("timestamp", new Date().getTime());
		responseParams.put("success", false);
		responseParams.put("code", status);
		response.setContentType("application/json;charset=UTF-8");
		responseParams.putAll(params);
		new ObjectMapper().writeValue(response.getOutputStream(), responseParams);
	}
}

File: src/main/kotlin/com/softwareplace/authorization/ResponseRegister.kt
package com.softwareplace.authorization

import com.fasterxml.jackson.databind.ObjectMapper
import com.softwareplace.authorization.JWTAuthenticationFilter.Companion.JWT
import com.softwareplace.json.logger.log.JsonLog
import com.softwareplace.json.logger.log.jsonLog
import com.softwareplace.json.logger.log.logger
import java.io.IOException
import java.time.LocalDateTime
import java.util.*
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

object ResponseRegister {

    @Throws(IOException::class)
    fun register(request: HttpServletRequest, response: HttpServletResponse) =
        register(request, response, "Access denied!", response.status, HashMap())

    @JvmStatic
    @Throws(IOException::class)
    fun register(request: HttpServletRequest, response: HttpServletResponse, message: String?, status: Int, params: Map<String, Any>): JsonLog {
        val responseParams = HashMap<String?, Any?>()
        val responseMessage = message ?: "Unexpected Error"
        responseParams["message"] = responseMessage
        responseParams["timestamp"] = Date().time
        responseParams["success"] = false
        responseParams["code"] = status
        response.status = status
        response.contentType = "application/json;charset=UTF-8"
        responseParams.putAll(params)
        ObjectMapper().writeValue(response.outputStream, responseParams)

        val logParams = responseParams.filter { (k, _) -> k != JWT }

        return logger.jsonLog
            .message(responseMessage)
            .add("status", status)
            .add("service", request.requestURI)
            .add("date", LocalDateTime.now())
            .add("customProperties", logParams)
    }
}
